jlm/util/BiMap.hpp

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2023 Håvard Krogstie <krogstie.havard@gmail.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#ifndef JLM_UTIL_BI_MAP_HPP
+#define JLM_UTIL_BI_MAP_HPP
+
+#include <jlm/util/common.hpp>
+
+#include <unordered_map>
+
+namespace jlm::util
+{
+/**
+ * Represents a bijective mapping between keys and values, with fast loopups by both key and value.
```

Comment:
> with fast loopups
should be changed to "with fast lookups". Could you be more precise here and specify what fast means, i.e., amortized constant time?